### PR TITLE
update ed25519-dalek and rand

### PR DESCRIPTION
**Pull request by Geal**
This PR contains breaking changes due to the `rand` update and will require being released in a major version update

### CODE REVIEW

**Comment by divarvel** [comment]
the commit where i regenerate samples is broken since apparently token verification times out on my machine :-(

**Comment by Geal** [comment]
updated to RC.2, everything good so far. Now we just need to wait for the release https://github.com/dalek-cryptography/ed25519-dalek/issues/245

File: biscuit-auth/Cargo.toml
Comment (Geal) [reply biscuit-auth/Cargo.toml:40]: YES YES YES awesome!

**Comment by pinkforest** [comment]
yay CI agrees :heavy_check_mark:  :partying_face:

File: biscuit-auth/Cargo.toml
Comment (pinkforest) [reply biscuit-auth/Cargo.toml]: yay even CI agrees :heavy_check_mark:  :partying_face:

**Review by divarvel — APPROVED** [review]

**Comment by divarvel** [comment]
(imo this could be squash-merged, the intermediate steps are not useful in this context)

**Comment by Geal** [comment]
after merging the latest changes from main that use static external keys in samples, I can now confirm that it is compatible both ways: main can validate tokens from this branch, and this branch validates tokens from main :partying_face:

**Pull request merged**
